signer/src/emily_client.rs

```diff
@@ -57,30 +89,75 @@ impl TryFrom<&Url> for EmilyClient {
         }
 
         let mut config = EmilyApiConfig::new();
-        config.base_path = url.to_string();
+        // Url::parse defaults `path` to `/` even if the parsed url was without the trailing `/`
+        // causing the api calls to have two leading slashes in the path (getting a 404)
+        config.base_path = url.to_string().trim_end_matches("/").to_string();
 
         Ok(Self { config })
     }
 }
 
 impl EmilyInteract for EmilyClient {
     async fn get_deposits(&self) -> Result<Vec<CreateDepositRequest>, Error> {
-        // just to kill the unused warning for now (Self.config)
-        let _ = &self.config;
-
-        // TODO: We need to be able to build `CreateDepositRequests` from the `DepositInfo` response.
-        // However, we don't have all of the information we need yet from Emily to do that.
-        // For now, we'll just return an empty vector.
-        Ok(vec![])
-
-        // let _resp = deposit_api::get_deposits(
-        //         &self.config,
-        //         Status::Pending,
-        //         None,
-        //         None
-        //     )
-        //     .await
-        //     .map_err(EmilyClientError::GetDeposits)?;
+        let resp = deposit_api::get_deposits(&self.config, Status::Pending, None, None)
+            .await
+            .map_err(EmilyClientError::GetDeposits)
+            .map_err(Error::EmilyApi)?;
+
+        // TODO: fetch multiple pages?
+        resp.deposits
```

Comment:
The `page_size` value seems to be passed to the emily api endpoint, using `None` should mean _no limit_ since it's used [here](https://github.com/stacks-network/sbtc/blob/main/emily/handler/src/database/entries/mod.rs#L251), but at the same time there's seems to be a "[forced pagination](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Query.Pagination.html)" once the resulting data reach 1MB.